raiden-ts/src/migration/3.ts

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { AddressZero, Zero, HashZero } from 'ethers/constants';
+import findKey from 'lodash/findKey';
+
+import { SignatureZero } from '../constants';
+
+const emptyBP = {
+  chainId: Zero,
+  tokenNetworkAddress: AddressZero,
+  channelId: Zero,
+  nonce: Zero,
+  transferredAmount: Zero,
+  lockedAmount: Zero,
+  locksroot: HashZero,
+  additionalHash: HashZero,
+  signature: SignatureZero,
+};
+
+function migrateChannelEnd(channel: any, endsAddr: any) {
+  for (const [prop, addr] of Object.entries<any>(endsAddr)) {
+    if (channel[prop].balanceProof) {
+      Object.assign(channel[prop].balanceProof, {
+        additionalHash: channel[prop].balanceProof.messageHash,
+      });
+      delete channel[prop].balanceProof.messageHash;
+      delete channel[prop].balanceProof.sender;
+    } else {
+      Object.assign(channel[prop], { balanceProof: emptyBP, locks: [] });
+    }
+    Object.assign(channel[prop], { address: addr });
+    if (!('withdraw' in channel[prop])) {
+      Object.assign(channel[prop], { withdraw: Zero });
+    }
+  }
+}
+
+/**
+ * Adds: Channel.closeParticipant, TransferState.partner & RaidenState.received
+ *
+ * @param state - RaidenState version 2
+ * @returns State version 3
+ */
+export default function migrate3(state: any) {
+  const newChannels = {};
+  for (const [tokenNetwork, partnerChannels] of Object.entries<any>(state.channels)) {
+    for (const [partner, channel] of Object.entries<any>(partnerChannels)) {
+      migrateChannelEnd(channel, { own: state.address, partner });
+      // rename messageHash to additionalHash, remove unused sender from BP
+      const token = findKey(state.tokens, (tn) => tn === tokenNetwork);
+      Object.assign(channel, { token, tokenNetwork });
+      Object.assign(newChannels, { [`${partner}@${tokenNetwork}`]: channel });
+    }
+  }
+  Object.assign(state, { channels: newChannels, oldChannels: {} });
```

Comment:
Before this PR we would not keep the old channels anywhere right?

Reply:
Right, settled channels were `unset` and forgotten. It's not used nor I expect to, but it may be convenient to not lose anything and save for later if needed. Channel state is not big